pkg/cli/service.go

```diff
@@ -123,6 +134,198 @@ func getImage(ctx context.Context, config searchConfig, username, password, imag
 	}
 }
 
+func (service searchService) getImagesByCveID(ctx context.Context, config searchConfig, username,
+	password, cveID string, c chan stringResult, wg *sync.WaitGroup) {
+	defer wg.Done()
+	defer close(c)
+
+	query := fmt.Sprintf(`{ImageListForCVE(id: "%s") {`+`
+								Name Tags }
+						  }`,
+		cveID)
+	result := &imagesForCve{}
+
+	endPoint, err := combineServerAndEndpointURL(*config.servURL, "/query")
+	if err != nil {
+		if isContextDone(ctx) {
+			return
+		}
+		c <- stringResult{"", err}
+
+		return
+	}
+
+	err = makeGraphQLRequest(endPoint, query, username, password, *config.verifyTLS, result)
+	if err != nil {
+		if isContextDone(ctx) {
+			return
+		}
+		c <- stringResult{"", err}
+
+		return
+	}
+
+	if result.Errors != nil {
+		var errBuilder strings.Builder
+
+		for _, err := range result.Errors {
+			fmt.Fprintln(&errBuilder, err.Message)
+		}
+
+		if isContextDone(ctx) {
+			return
+		}
+		c <- stringResult{"", errors.New(errBuilder.String())} //nolint: goerr113
+
+		return
+	}
+
+	var localWg sync.WaitGroup
+
+	p := newSmoothRateLimiter(ctx, &localWg, c)
+	localWg.Add(1)
+
+	go p.startRateLimiter()
+
+	for _, image := range result.Data.ImageListForCVE {
+		for _, tag := range image.Tags {
+			localWg.Add(1)
+
+			go addManifestCallToPool(ctx, config, p, username, password, image.Name, tag, c, &localWg)
+		}
+	}
+
+	localWg.Wait()
+}
+
+func (service searchService) getImageByNameAndCVEID(ctx context.Context, config searchConfig, username,
```

Comment:
Is it duplicate of getImagesByCveID

Reply:
No. This one I have added on cli side without a corresponding API for it

If the user types an image name AND a cve id, it will print all the tags which are affected by that image. And if the user adds `--fixed` option on top of it, it will return the output from fixed tags API.

I added this "intermediate" combination so the CLI feels more intuitive. I am basically just filtering the list of images by image name in results of "imageByCVEID"

Reply:
Can you find a way to provide that functionality by re-using or changing imageByCVEID method, it will reduce our code base and increase our test code coverage